### PR TITLE
Topic tile view

### PR DESCRIPTION
**Pull request by matthewfeinberg**
Ticket # 10 Landing Topic Tile View. Created topic tiles, fav icon, sorts, and new routes.

### CODE REVIEW

File: client/src/__tests__/App.test.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 // this adds custom jest matchers from jest-dom
-import 'jest-dom/extend-expect'
-import App from '../App';
+import "jest-dom/extend-expect";
+import App from "../App";
 
 afterEach(cleanup);
 
+it("CheckboxWithLabel changes the text after click", async () => {
+  const { getByText } = render(<App />);
```
Comment (NicholasMiron) [inline client/src/__tests__/App.test.js]: Incomplete test. It doesn't look like the function closes. Causes tests to fail

File: client/src/topicTiles.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { withStyles } from "@material-ui/core/styles";
+import Grid from "@material-ui/core/Grid";
+import GridList from "@material-ui/core/GridList";
+import GridListTile from "@material-ui/core/GridListTile";
+import GridListTileBar from "@material-ui/core/GridListTileBar";
+import IconButton from "@material-ui/core/IconButton";
+import Card from "@material-ui/core/Card";
+import CardHeader from "@material-ui/core/CardHeader";
+import CardMedia from "@material-ui/core/CardMedia";
+import CardContent from "@material-ui/core/CardContent";
```
Comment (NicholasMiron) [inline client/src/topicTiles.js]: Are all of these imports needed? If not can we only bring in what is needed?

File: client/src/App.js [hunk 3]
```diff
@@ -93,30 +144,47 @@ export default class App extends Component {
...
-    if (this.state.page === 'home') {
+    if (this.state.page === "home") {
       return (
         <>
           <SearchAppBar toggleModal={this.toggleModal} />
-          <LandingPage topics={[]} toggleModal={this.toggleModal}/>
-          <Modal modalType={this.state.modalType} isOpen={this.state.isOpen} toggleOpen={this.toggleModal} setLogin={this.setLoginState}/>
-          <button name="action" onClick={(e) => this.handlePageChange(e)}>Go To Action Page</button>
+          <LandingPage topics={[]} toggleModal={this.toggleModal} />
+          <TopicTiles
```
Comment (NicholasMiron) [inline client/src/App.js]: This might need to go into the landing page component. Will discuss as team.

**Review by NicholasMiron — CHANGES_REQUESTED** [review]

**Comment by matthewfeinberg** [comment]
moved Topic Tiles from App.js to LandingPage.js

File: client/src/topicTiles.js (hunk 2 above)
Comment (matthewfeinberg) [reply client/src/topicTiles.js]: removed unnecessary imports

File: client/src/__tests__/App.test.js (hunk 1 above)
Comment (matthewfeinberg) [reply client/src/__tests__/App.test.js]: this was done by someone else, Austin and I fixed it

File: client/src/App.js (hunk 3 above)
Comment (matthewfeinberg) [reply client/src/App.js]: moved content to the landing page

**Review by NicholasMiron — APPROVED** [review]

**Review by NicholasMiron — CHANGES_REQUESTED** [review]
Looks like you still have merge conflicts

File: client/src/App.js [hunk 4]
```diff
@@ -184,6 +191,12 @@ export default class App extends Component {
           <button name="home" onClick={e => this.handlePageChange(e)}>
             Go To Home Page
           </button>
+=======
+          <SearchAppBar toggleModal={this.toggleModal} />
+          <TopicPageContainer currentTopic={this.state.currentTopic} />
+          <Modal modalType={this.state.modalType} isOpen={this.state.isOpen} toggleOpen={this.toggleModal} setLogin={this.setLoginState} />
+          <button name="home" onClick={(e) => this.handlePageChange(e)}>Go To Home Page</button>
+>>>>>>> e4295b61458106a54a3001f3814b218b1faeadbe
```
Comment (NicholasMiron) [inline client/src/App.js]: Merge conflicts in App

File: client/src/Components/LandingPage.js [hunk 5]
```diff
@@ -3,22 +3,23 @@ import PropTypes from "prop-types";
...
-        handleTopicTileClick={props.handleTopicTileClick}
-        favoritedTopics={props.favoritedTopics}
-      />
+      <Grid style={{ paddingBottom: 5 }}>
+        <TopicTiles
+          allTopics={props.allTopics}
+          handleTopicTileClick={props.handleTopicTileClick}
+          favorites={props.favorites}
+        />
+        <HeroBanner name={"Octavius"} />
```
Comment (NicholasMiron) [inline client/src/Components/LandingPage.js]: Why did the hero banner get added here as well? it should have already been brought in.

File: client/src/__tests__/App.test.js [hunk 6]
```diff
@@ -23,6 +28,11 @@ afterEach(cleanup);
 //     expect(dolphin).toBeTruthy();
 // });
 
+<<<<<<< HEAD
 xit("Is a dummy test", async () => {
   expect(true).toBe(true);
+=======
+xit('Is a dummy test', async () => {
+    expect(true).toBe(true);
+>>>>>>> e4295b61458106a54a3001f3814b218b1faeadbe
```
Comment (NicholasMiron) [inline client/src/__tests__/App.test.js]: Unresolved merge conflicts

File: db/db.js [hunk 7]
```diff
@@ -31,9 +37,21 @@ const getHashedPassword = (login, cb) => {
...
+
+    pool.query(`select * from users where username = '${login.username}'`, (err, data) => {
+        if (err) {
+            cb(err);
+        }
+        else {
+            cb(null, data)
+        }
+    })
+>>>>>>> e4295b61458106a54a3001f3814b218b1faeadbe
```
Comment (NicholasMiron) [inline db/db.js]: Unresolved merge conflicts

File: db/db.js [hunk 8]
```diff
@@ -131,6 +202,20 @@ const handleSignup = (userInfo, cb) => {
...
+            console.log('IT FAILED!', err)
+            cb(err)
+        }
+        else {
+            cb(null, data)
+        }
+    })
+}
+
+>>>>>>> e4295b61458106a54a3001f3814b218b1faeadbe
```
Comment (NicholasMiron) [inline db/db.js]: Unresolved merge conflicts

**Review by NicholasMiron — CHANGES_REQUESTED** [review]

**Review by NicholasMiron — CHANGES_REQUESTED** [review]
Please resolve issues and we will get this approved.

**Review by NicholasMiron — APPROVED** [review]

**Pull request merged**